draft-ietf-httpbis-message-signatures.md

```diff
@@ -382,7 +343,153 @@ This example shows a canonicalized value for the parameters of a given signature
   created=1618884475;expires=1618884775
 ~~~
 
-Note that an HTTP message could contain multiple signatures, but only the signature parameters used for the current signature are included in this field.
+Note that an HTTP message could contain multiple signatures, but only the signature parameters used for the current signature are included in this entry.
+
+### Method {#content-request-method}
+
+The `@method` identifier refers to the HTTP method of a request message. The value of is canonicalized by taking the value of the method as a string. Note that the method name is case-sensitive as per {{!SEMANTICS=RFC7231}}, and conventionally standardized method names are uppercase US-ASCII.
+
+For example, the following request message:
+
+~~~
+POST /path?param=value HTTP/1.1
+Host: www.example.com
+~~~
+
+Would result in the following `@method` value:
+
+~~~
+"@method": POST
+~~~
+
+### Authority {#content-request-authority}
+
+The `@authority` identifier refers to the authority component of the target of the HTTP request message. In HTTP 1.1, this is usually conveyed using the `Host` header, while in HTTP 2 and HTTP 3 it is conveyed using the `:authority` pseudo-header. The value is the fully-qualified authority component of the request, comprised of the host and, optionally, port of the request target, as a string.
+
+For example, the following request message:
+
+~~~
+POST /path?param=value HTTP/1.1
+Host: www.example.com
+~~~
+
+Would result in the following `@authority` value:
+
+~~~
+"@authority": www.example.com
+~~~
+
+### Scheme {#content-request-scheme}
+
+The `@scheme` identifier refers to the scheme of the target URL of the HTTP request message. The value is the scheme as a string.
+
+For example, the following request message requested over plain HTTP:
+
+~~~
+POST /path?param=value HTTP/1.1
+Host: www.example.com
+~~~
+
+Would result in the following `@scheme` value:
+
+~~~
+"@scheme": http
+~~~
+
+### Request Origin {#content-request-origin}
+
+The `@request-origin` identifier refers to the full origin component of the HTTP request message. The value is combination of the absolute path and query components of the request URL.
+
+For example, the following request message:
+
+~~~
+POST /path?param=value HTTP/1.1
+Host: www.example.com
+~~~
+
+Would result in the following `@request-origin` value:
+
+~~~
+"@request-origin": /path?param=value
+~~~
+
+The following OPTIONS request message:
+
+~~~
+OPTIONS * HTTP/1.1
+Host: www.example.com
+~~~
+
+Would result in the following `@request-origin` value:
+
+~~~
+"@request-origin": *
+~~~
+
+### Path {#content-request-path}
+
+The `@path` identifier refers to the target path of the HTTP request message. The value is the absolute path of the request target, with no query parameters and no trailing `?` character.
```

Comment:
Best to use "query component" throughout (and maybe reference RFC3986); they're not always parameterised in the `a=b&c=d` sense.

Reply:
Good point, will do.